doc/design/embargo-mvp.md

```diff
@@ -0,0 +1,131 @@
+
+* Get/List dandiset endpoint
+
+  The DandisetViewSet queryset will filter out dandisets with `embargo_status == PRIVATE` that are also not owned by the current user.
+  This will prevent them from showing up in the listing or fetching endpoints.
+
+  The `DandisetSerializer` will also be updated to include the `embargo_status` field so that the web client can render the embargoed dandiset appropriately.
+
+* publish version endpoint
+
+  Return error 400 if `dandiset.embargo_status != PUBLIC`.
+
+* create asset, update metadata, and any other dandiset/version modification endpoints:
+
+  Return error 400 if `dandiset.embargo_status == RELEASING`.
+
+* New endpoint: `POST /api/dandisets/{dandiset_id}/release`
+
+  Release an embargoed dandiset.
+  
+  Only permitted for owners and admins. If the `embargo_status` is `PUBLIC` or `RELEASING`, return 400.
+
+  Set the `embargo_status` to `RELEASING`, then dispatch the release task.
+
+* Release task
+
+  For every `Asset` with an `EmbargoedAssetBlob` in the dandiset, convert the `EmbargoedAssetBlob` into an `AssetBlob` by moving the data from the embargo bucket to the public bucket.
+  These could be >5GB, so the [multipart copy API](https://docs.aws.amazon.com/AmazonS3/latest/userguide/CopyingObjectsMPUapi.html) must be used.
+  Once finished, the `access` metadata field on the dandiset will be updated to `OpenAccess` and `embargo_status` is set to `PUBLIC`.
+  
+  Before copying data, check if an existing `AssetBlob` with the same checksum has been uploaded already (this would have happened after uploading the embargoed data).
+  If so, use it instead of copying the `EmbargoedAssetBlob` data.
+
+* Get/List asset endpoint
+
+  The AssetViewSet queryset will filter out assets with `embargoed_asset_blob.dandiset.embargo_status != PUBLIC` that are also not owned by the current user.
+  This will prevent them from showing up in the listing or fetching endpoints.
+
+* upload_initialize_view
+
+  An optional field `embargoed_dandiset` will be available on the serializer.
+  If specified, it will be passed to the `Upload` object.
+  This will mean the final upload will result in an `EmbargoedAssetBlob`.
+
+  Even if `embargoed_dandiset` is specified, if an `AssetBlob` with a matching checksum already exists, return it instead of uploading embargoed data.
+  This means that an embargoed dandiset can contain both embargoed and unembargoed assets.
+
+  `EmbargoedAssetBlob`s should also be checked for deduplication, but only within an embargoed dandiset.
+  An embargoed dandiset should use the same `EmbargoedAssetBlob` if the same file appears in multiple places, but two embargoed dandisets should upload the same data twice if they both contain the same file.
+
+  Return error 400 if `dandiset.embargo_status == RELEASING`.
+
+* TODO zarr upload
+
+* stats_view
+  
+  The total size value should include the size of `EmbargoedAssetBlob`s as well as `AssetBlob`s.
```

Comment:
For the MVP, let's not worry about this. I think in the full version, it would be nice to see something like "300 TB published (of which 100 TB currently under embargo)" with `embargo` linked to an explanation.